### PR TITLE
[NC]: Tab selector v2

### PR DESCRIPTION
**Pull request by nicolaschapur**
This PR includes a new react component for the tabs selector, with the new UI for the dropdown. It also includes close button functionality, new title and checkbox removed

https://github.com/Mocksi/HARlighter/assets/95246792/f808df91-2ef7-40b1-9104-bcbc0e69a2d5

## Summary by CodeRabbit

- **New Features**
  - Introduced a `TabSelector` component for selecting browser tabs using a dropdown menu.
  - Added functionality to render the `TabSelector` component on the web page.

- **Style**
  - Updated styles for dropdown menus, including hover effects and separators.

- **Refactor**
  - Changed the initialization of `demos` to an empty array in the `CreateDemo` component.
  - Modified `tab-selector.html` to reference `tab-selector.tsx` for rendering the `TabSelector` component.

### CODE REVIEW

File: apps/mocksi-lite/pages/tab-selector.html [hunk 1]
```diff
@@ -5,26 +5,7 @@
...
-                </select>
-            </label>
-            <div class="tab-selector-checkbox">
-                <input type="checkbox" />
-                <div>Highlight All Previous Changes</div>
-            </div>
-        </div>
-        <script src="tab-selector.ts" type="module"></script>
+    <body id="tab-selector-container">
+        <script src="tab-selector.tsx"></script>
```
[the remark this answers is not in the document]
Comment (nicolaschapur) [reply apps/mocksi-lite/pages/tab-selector.html:9]: It seems to be working fine in prod mode!
https://github.com/Mocksi/HARlighter/assets/95246792/2ec46ae2-3ead-4680-9ba7-08944b6c3a4f

File: apps/mocksi-lite/pages/TabSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import chevronDown from "../public/chevron-down.png";
+
+const TabSelector = () => {
+	const [tabs, setTabs] = useState<chrome.tabs.Tab[]>([]);
+	const [selected, setSelected] = useState<string>("Select");
+	const [open, setOpen] = useState(false);
+	// const activeTab = tabs.find(t => t.active);
+
+	useEffect(() => {
+		chrome.tabs.query({}, (result) => setTabs(result));
```
Comment (elg0nz) [inline apps/mocksi-lite/pages/TabSelector.tsx:12]: Small favor, could you try `chrome.tabs.query({{ active: true, lastFocusedWindow: true }, (result) => setTabs(result));` ?

Comment (elg0nz) [inline apps/mocksi-lite/pages/TabSelector.tsx:12]: nit: It would also be great to filter out the `mocksi register` from the tab results.

**Review by elg0nz — APPROVED** [review]
Great stuff!
Re the .tsx comment, I also think it's fine. We'll just double check things when we create the next build. Thanks!!

File: apps/mocksi-lite/pages/TabSelector.tsx (hunk 2 above)
Comment (nicolaschapur) [reply apps/mocksi-lite/pages/TabSelector.tsx:12]: I used it yesterday to test the query and definitely works to get the last active tab

**Pull request merged**